idea/src/org/jetbrains/kotlin/idea/inspections/logging/KotlinLoggerInitializedWithForeignClassInspection.kt

```diff
@@ -0,0 +1,157 @@
+            )
+        )
+        return UiUtils.createAddRemoveTreeClassChooserPanel(table, KotlinBundle.message("choose.logger.factory.class"))
+    }
+
+    override fun readSettings(element: Element) {
+        super.readSettings(element)
+        BaseInspection.parseString(loggerFactoryClassName, loggerFactoryClassNames)
+        BaseInspection.parseString(loggerFactoryMethodName, loggerFactoryMethodNames)
+        if (loggerFactoryClassNames.isEmpty() || loggerFactoryClassNames.size != loggerFactoryMethodNames.size) {
+            BaseInspection.parseString(DEFAULT_LOGGER_FACTORY_CLASS_NAME, loggerFactoryClassNames)
+            BaseInspection.parseString(DEFAULT_LOGGER_FACTORY_METHOD_NAME, loggerFactoryMethodNames)
+        }
+    }
+
+    override fun writeSettings(element: Element) {
+        loggerFactoryClassName = BaseInspection.formatString(loggerFactoryClassNames)
+        loggerFactoryMethodName = BaseInspection.formatString(loggerFactoryMethodNames)
+        XmlSerializer.serializeInto(this, element, object : SerializationFilterBase() {
+            override fun accepts(accessor: Accessor, bean: Any, beanValue: Any?): Boolean {
+                if (accessor.name == "loggerFactoryClassName" && beanValue == DEFAULT_LOGGER_FACTORY_CLASS_NAME) return false
+                if (accessor.name == "loggerFactoryMethodName" && beanValue == DEFAULT_LOGGER_FACTORY_METHOD_NAME) return false
+                return true
+            }
+        })
+    }
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) = dotQualifiedExpressionVisitor(fun(dotQualified) {
+        val containingClass = dotQualified.containingClass() ?: return
+        val containingClassName = containingClass.name ?: return
+
+        val call = dotQualified.callExpression ?: return
+        val callee = call.calleeExpression ?: return
+        val loggerMethodFqNames = loggerFactoryFqNames[callee.text] ?: return
+
+        val argument = call.valueArguments.singleOrNull()?.getArgumentExpression() as? KtDotQualifiedExpression ?: return
+        val argumentSelector = argument.selectorExpression ?: return
+        val classLiteral = when (val argumentReceiver = argument.receiverExpression) {
+            // Foo::class.java, Foo::class.qualifiedName
+            is KtClassLiteralExpression -> {
+                val selectorText = argumentSelector.safeAs<KtNameReferenceExpression>()?.text
+                if (selectorText != "java" && selectorText != "qualifiedName") return
+                argumentReceiver
+            }
+            // Foo::class.java.name, Foo::class.java.getName()
+            is KtDotQualifiedExpression -> {
+                val classLiteral = argumentReceiver.receiverExpression as? KtClassLiteralExpression ?: return
+                if (argumentReceiver.selectorExpression.safeAs<KtNameReferenceExpression>()?.text != "java") return
+                if (argumentSelector.safeAs<KtNameReferenceExpression>()?.text != "name" &&
```

Comment:
There are also: "getSimpleName", "getCanonicalName". I see that Java inspection doesn't check those functions. But me for it's rather Java inspection bug than feature.